golang/x/relay/keeper/validator.go

```diff
@@ -0,0 +1,91 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/summa-tx/relays/golang/x/relay/types"
+)
+
+func (k Keeper) getConfs(ctx sdk.Context, header types.BitcoinHeader) (uint32, sdk.Error) {
+	bestKnown, err := k.GetBestKnownDigest(ctx)
+	if err != nil {
+		return 0, err
+	}
+	bestKnownHeader, err := k.GetHeader(ctx, bestKnown)
+	if err != nil {
+		return 0, err
+	}
+	return bestKnownHeader.Height - header.Height, nil
+}
+
+func (k Keeper) validateProof(ctx sdk.Context, proof types.SPVProof, requestID types.RequestID) sdk.Error {
```

Comment:
Can you elaborate?  Not sure I understand what you're asking.

Reply:
`validateProof` loads a Request and checks confs inside the function

As a result, `checkRequestsFilled` checks the confs of the first request outside a loop, then the rest _inside_ the loop

Can we have `validateProof` return the number of confs, so that all checks are _inside_ the loop?

Reply:
Ok, I understand what you are saying.  Except I think it makes more sense for us to get confs inside `checkRequestsFilled` instead of getting it through ValidateProof. But it will still work to simplify the for loop.